dlp/snippets/inspect/inspect_test.go

```diff
@@ -681,3 +633,218 @@ func TestInspectTableWithCustomHotword(t *testing.T) {
 		t.Errorf("TestInspectTableWithCustomHotword got %q, want %q", got, want)
 	}
 }
+
+func TestInspectDataStoreSendToScc(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	var buf bytes.Buffer
+	u := uuid.New().String()[:8]
+	datastoreNamespace := "golang-samples" + u
+	datastoreKind := "task"
+
+	if err := inspectDataStoreSendToScc(&buf, tc.ProjectID, datastoreNamespace, datastoreKind); err != nil {
+		t.Fatal(err)
+	}
+
+	got := buf.String()
+	if want := "Job created successfully:"; !strings.Contains(got, want) {
+		t.Errorf("InspectBigQuerySendToScc got %q, want %q", got, want)
+	}
+}
+
+func TestInspectGCSFileSendToScc(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	var buf bytes.Buffer
+
+	gcsPath := fmt.Sprint("gs://" + bucketNameForInspectGCSSendToScc + "/" + dirPathForInspectGCSSendToScc + "test.txt")
+
+	if err := inspectGCSFileSendToScc(&buf, tc.ProjectID, gcsPath); err != nil {
+		t.Fatal(err)
+	}
+
+	got := buf.String()
+	if want := "Job created successfully:"; !strings.Contains(got, want) {
+		t.Errorf("TestInspectGCSFileSendToScc got %q, want %q", got, want)
+	}
+}
+
+func TestMain(m *testing.M) {
+	tc := testutil.Context{}
+	tc.ProjectID = os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	if tc.ProjectID == "" {
+		tc.ProjectID = os.Getenv("")
+	}
+	setup(tc.ProjectID)
+
+	exitVal := m.Run()
+
+	buckets := []string{bucketNameForInspectGCSSendToScc, bucketnameForInspectGCSFileWithSampling}
+	for _, v := range buckets {
+		deleteBucket(tc.ProjectID, v)
+	}
+	os.Exit(exitVal)
+}
+
+func setup(projectID string) {
+	filePathtoGCS(projectID)
+	createBucket(projectID)
+}
+
+func createBucket(projectID string) error {
```

Comment:
question: why doesn't this use `testutil.CleanBucket()`?

https://github.com/GoogleCloudPlatform/golang-samples/blob/main/internal/testutil/storage_cleaner.go#L40

Reply:
Hey @telpirion, 
Hii,
I wanted to clarify my previous comment, as I'm concerned it might not have been entirely clear. In response to your initial suggestion, I proceeded by implementing the `TestMain` function as you advised. Additionally, I explored the functions within the `testutil` package, specifically within the `storage_cleaner.go` file.

as I mentioned in https://github.com/GoogleCloudPlatform/golang-samples/pull/3222#discussion_r1284235043 I encountered some challenges when working with both approaches. While attempting to call the creation and deletion functions of a bucket from the `TestMain` function, I found that these functions required a variable of type `t *testing.T` to be passed. However, this wasn't feasible using the `TestMain` approach.

In light of this, I considered using a custom function to handle the creation and cleaning of the bucket, which I had implemented previously. However, as you mentioned, the `testutil` package includes the `storage_cleaner.go` file, which contains functions for creating and cleaning the bucket.

Regarding the latest commit:

I refrained from using `testutil.Cleanbucket()` due to its return of the `cleanBucketWithClient()` function. This function waits for the `bucketExists` condition, which results in a situation where one bucket consistently exists in the project. As per my understanding, my current implementation is effective: it generates a bucket at the test's inception with a unique name, and subsequently removes the bucket using the `DeleteBucketIfExists` function upon test completion.
Please let me know if this explanation provides a clearer understanding of the recent changes I've made, please let me know if are we on the same page or not, and if not could you please tell me how should i proceed further. Thank You and apologize that it's kind of taking somewhat more time and efforts.

Reply:
Using one-off, custom bucket creation/deletion functions creates a maintenance burden for repo maintainers. If having the bucket setup & teardown in `TestMain()` isn't feasible, then it is okay to have it in the test function.

Reply:
Hey @telpirion, Thank You for the reply, So that's why I have used functions from `testutil` package here on line no. 676  for creation of bucket, https://github.com/GoogleCloudPlatform/golang-samples/blob/e23ad7f34e1e61a6ab9d4d7130045294576aceac/dlp/snippets/inspect/inspect_test.go#L676 and on line no 696 for deletion of bucket https://github.com/GoogleCloudPlatform/golang-samples/blob/e23ad7f34e1e61a6ab9d4d7130045294576aceac/dlp/snippets/inspect/inspect_test.go#L696.
Though let me know if any further changes are required. Thank You